### PR TITLE
Add bay water features

### PR DESCRIPTION
**Pull request by rmarianski**
Refs #400

@zerebubuth could you review please?

### CODE REVIEW

**Comment by rmarianski** [comment]
Just remembered that I should probably look to add a test and upgrade the migration for the next release before you review.

**Comment by zerebubuth** [comment]
I forget - do we need a migration to do a fake update for `natural IN ('bay', 'strait', 'fjord')` to make the index pick up these new values, or do we need to rebuild the index?

**Comment by rmarianski** [comment]
In the past I think we've just been rebuilding the entire index by creating a new one and then in a transaction dropping the old and renaming the new.

A fake update sounds like it might be more targeted though. What would that look like? `update planet_osm_polygon set "natural"="natural" where "natural" in ('bay', 'strait', 'fjord')`?

**Comment by zerebubuth** [comment]
Yeah, I _think_ something like that would work, but I'm not 100% sure. Would be a good idea to test it (perhaps on dev) to make sure that it does what we expect.

**Comment by nvkelso** [comment]
BRB: Add migration and test case.

**Comment by rmarianski** [comment]
Depends on https://github.com/mapzen/TileStache/pull/113

@zerebubuth could you review now please?

The targeted update for the index didn't work for me locally, so I added a migration to make a new index.

I transitioned how the `mz_is_building` property is being removed. The previous method was really specific, and that didn't seem necessary to me. Let me know if I missed something here though.

It turns out that I had to add more flexibility to the transform in order to check if `label_placement` was in the list of properties. I put in an additional `properties` variable in the eval scope to point to the original dict to accomplish that.

File: test/400-bay-water.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+assert_has_feature(
```
Comment (nvkelso) [inline test/400-bay-water.py]: Please add 2 additional tests for `strait` and `fjord`.

File: queries.yaml [hunk 2]
```diff
@@ -231,12 +237,11 @@ post_process:
       property_keys: [name, kind]
       geometry_types: [Point]
       min_distance: 256.0
-  - fn: TileStache.Goodies.VecTiles.transform.drop_features_where
+  - fn: TileStache.Goodies.VecTiles.transform.drop_properties
```
Comment (zerebubuth) [inline queries.yaml]: This filter was supposed to drop any building polygons from the `landuse` layer, but will now just drop the `mz_is_building` property and keep the feature. Perhaps this should be changed to?

```yaml
  - fn: TileStache.Goodies.VecTiles.transform.drop_features_where
    params:
      source_layer: landuse
      where: >-
        properties.get('mz_is_building') == True
```

Comment (rmarianski) [inline queries.yaml]: Nice catch! I thought the intent actually was to only drop the property. I'll go ahead and correct this.

Comment (rmarianski) [inline queries.yaml]: Corrected in d6fbfd8.

**Comment by zerebubuth** [comment]
:+1: looks great!

**Pull request merged**